api/v1alpha7/conversion.go

```diff
@@ -63,6 +63,11 @@ var v1alpha8OpenStackClusterRestorer = conversion.RestorerFor[*infrav1.OpenStack
 			return &c.Status.Bastion.ReferencedResources
 		},
 	),
+	"allNodesSecurityGroup": conversion.UnconditionalFieldRestorer(
+		func(c *infrav1.OpenStackCluster) **infrav1.SecurityGroup {
+			return &c.Status.AllNodesSecurityGroup
```

Comment:
@mdbooth can you explain how you envision the upgrade path where we add the Calico rules during conversion? Will that happen in these conversion.go? I haven't seen it before in CAPO I think.